game/src/edit/roads.rs

```diff
@@ -0,0 +1,80 @@
+use map_gui::tools::PromptInput;
+use map_model::{Direction, LaneType, Road};
+use widgetry::{EventCtx, State};
+
+use crate::app::{App, Transition};
+use crate::edit::apply_map_edits;
+
+/// Specify the lane types for a road using a text box. This is a temporary UI to start
+/// experimenting with widening roads. It'll be replaced by a real UI once the design is ready.
+pub fn prompt_for_lanes(ctx: &mut EventCtx, road: &Road) -> Box<dyn State<App>> {
+    let r = road.id;
+    PromptInput::new(
+        ctx,
+        "Define lanes_ltr",
+        lanes_to_string(road),
+        Box::new(move |string, ctx, app| {
+            // We're selecting a lane before this, but the ID is probably about to be invalidated.
+            app.primary.current_selection = None;
+
+            let mut edits = app.primary.map.get_edits().clone();
+            edits.commands.push(app.primary.map.edit_road_cmd(r, |new| {
+                new.lanes_ltr = string_to_lanes(string.clone());
+            }));
+            apply_map_edits(ctx, app, edits);
+            Transition::Multi(vec![Transition::Pop, Transition::Pop])
+        }),
+    )
+}
+
+fn lanes_to_string(road: &Road) -> String {
+    // TODO Assuming driving on the right.
+    let mut dir_change = false;
+    let mut string = String::new();
+    for (_, dir, lt) in road.lanes_ltr() {
+        if !dir_change && dir == Direction::Fwd {
+            string.push('/');
+            dir_change = true;
+        }
+        string.push(
+            lane_type_codes()
+                .into_iter()
+                .find(|(x, _)| *x == lt)
+                .unwrap()
+                .1,
+        );
+    }
+    string
+}
+
+fn string_to_lanes(string: String) -> Vec<(LaneType, Direction)> {
+    let mut lanes = Vec::new();
+    let mut dir = Direction::Back;
+    for x in string.chars() {
+        if x == '/' {
+            dir = Direction::Fwd;
+            continue;
+        }
+        let lt = lane_type_codes()
+            .into_iter()
+            .find(|(_, code)| *code == x)
+            .unwrap()
+            .0;
+        lanes.push((lt, dir));
+    }
+    lanes
+}
+
+fn lane_type_codes() -> Vec<(LaneType, char)> {
+    vec![
+        (LaneType::Driving, 'd'),
+        (LaneType::Parking, 'p'),
+        (LaneType::Sidewalk, 's'),
+        (LaneType::Shoulder, 'S'),
+        (LaneType::Biking, 'b'),
+        (LaneType::Bus, 't'), // transit
+        (LaneType::SharedLeftTurn, 'l'),
```

Comment:
I suppose this is a shared _right_ turn in some locales... abbreviating things is even harder than naming things. 😓 

I submit my palette of colors for the bikeshed:

```diff
  (LaneType::Driving, 'd'),
  (LaneType::Parking, 'p'),
-  (LaneType::Sidewalk, 's'),
+ (LaneType::Sidewalk, 'w'),
  (LaneType::Shoulder, 'S'),
- (LaneType::Biking, 'b'),
+ (LaneType::Biking, 'c'), // cyclists with any number of wheels
- (LaneType::Bus, 't'), // transit
+ (LaneType::Bus, 'b'), 
-  (LaneType::SharedLeftTurn, 'l'),
+ (LaneType::SharedLeftTurn, 't'), 
- (LaneType::Construction, 'c'),
+ (LaneType::Construction, 'x'), // "x" as in "closed" - is that intuitive?
  (LaneType::LightRail, 'r'),
```

I'm not convinced it's better though. Feel free to keep your colors if you're not compelled.

Reply:
I literally hacked this "UI" together yesterday about an hour before Yuwen sent her designs. I'm happy to modify these abbreviations, but only we will be using this temporary string-based thing, and hopefully only for a few more days.